### ISSUE TITLE
’<’ not supported between instances of ‘str’ and 'int’ when editing settings

### ISSUE THREAD

**Issue by skynet01** [akasma74/Hass-Custom-Alarm#34]
**Your components (from Alarm/Settings/About):**
 - component version v1.10.1

**Platform & Browser:**
 - Platform name & Home assistant version [e.g. Hass.io 0.96.3]
 - Browser name  Chrome / Firefox

**Describe the bug**
When going to settings of the panel I get the following errors

Failed to call service alarm_control_panel/alarm_yaml_user. list indices must be integers or slices, not str
when saving any setting in the settings I get this
’<’ not supported between instances of ‘str’ and 'int’

**To Reproduce**
1. Go to settings pannel

**Expected behavior**
Saving settings 

**Screenshots**

**Additional context**
Installed using HACS
copied resources folder to /config
inside: /config/resources/bwalarm/bwalarm.yaml 
- platform: bwalarm

in the main confing
alarm_control_panel: !include resources/bwalarm/bwalarm.yaml

**Comment by akasma74** [comment]
Did you install the integration when on HA 0.96.3 or on previous version of HA that was updated later?

**Comment by skynet01** [comment]
With HA 0.96.3, it's a fresh install

**Comment by dnguyen800** [comment]
Hi, I also get the same error message as @skynet01 , but I have a slightly different HA setup. I cannot save any settings through the Alarm UI either. I did initially try to setup bwalarm settings in the bwalarm.yaml, but then erased and tried to save settings in the UI.

This is a first-time install of bwalarm. I have tried uninstalling and re-installing through HACS again but the issue remains.

**Your components (from Alarm/Settings/About):**
component version v1.10.1

**Platform & Browser:**
Platform name & Home assistant version [e.g. Hass.io 0.95.4]
**Browser name** Chrome 

**Describe the bug**
When pressing a button on the alarm (in the screenshot, I press the 'Away' button, I get this message:

`Failed to call service alarm_control_panel/alarm_arm_away`

when saving any setting in the settings I get this:

`’<’ not supported between instances of ‘str’ and 'int’`

**To Reproduce**
Steps to reproduce the behavior:

Click on any button in alarm panel, or change any setting.

**Expected behavior**
Saving settings

**Screenshots**
If applicable, add screenshots to help explain your problem.

![image](https://user-images.githubusercontent.com/25126347/62065499-f786a600-b1e3-11e9-81cf-de1d9797f3d1.png)


![image](https://user-images.githubusercontent.com/25126347/62065478-edfd3e00-b1e3-11e9-8ac3-3c948cda707f.png)


**Additional context**
Installed using HACS
copied resources folder to /config
inside: /config/resources/bwalarm/bwalarm.yaml

`platform: bwalarm`
in the main config
`alarm_control_panel: !include resources/bwalarm/bwalarm.yaml`

**Comment by dnguyen800** [comment]
Update: I installed bwalarm on another HA instance and it worked. The only difference I can think of is in the bwalarm.yaml file. I previously had this entry  (with the '-') in the yaml file:

```- platform: bwalarm```

Changing to this fixed the issue:

```platform: bwalarm```

Now that I see a working config, I can see why the dash was causing problems. The Alarm UI edits the bwalarm.yaml file and adds configuration entries. If you have another alarm besides bwalarm listed in the .yaml file, then you will run into the issue.

**Comment by akasma74** [comment]
@dnguyen800 so in your case it's another platform in bwalarm.yaml that created the issue, right?

**Comment by dnguyen800** [comment]
Correct, it was my fault as I wanted to include another MQTT alarm platform in bwalarm.yaml. By adding the '-', it messed up the YAML formatting when the UI alarm settings are changed.

**Comment by akasma74** [comment]
@skynet01 could you post content of your bwalarm.yaml here?

**Comment by skynet01** [comment]
Yes, I can confirm that removing the "-" from the config fixes the issue. No more errors in the setting and everything seems to be working. Would be nice to have this fixed since I have other devices that have built-in alarm panels like Arlo that I can't use

**Comment by akasma74** [comment]
@skynet01 do you know how to fix it?
as far as I understand, it's incorrect YAML syntax and there is nothing anyone can fix apart from correcting the yaml file itself - correct me if I'm wrong.

**Comment by skynet01** [comment]
Maybe the config for the app should be in a separate file since it’s managed by the ui anyway. That way we can just have our alarm.yaml file in the main config folder with just single line platform declaration.

**Comment by akasma74** [comment]
Got it. Need to think about it as it's a bit complicated stuff

**Comment by skynet01** [comment]
Can't you just have the component load with "platform: bwalarm" in alarm.yaml file and then just keep all of the config parameters in a separate settings file? Or its not that simple
